cpp/src/io/json/json_gpu.cu

```diff
@@ -189,14 +193,23 @@ __device__ field_descriptor next_field_descriptor(const char* begin,
 
           // Skip the colon between the key and the value
           auto const value_begin = thrust::find(thrust::seq, key_range.second, end, ':') + 1;
-          return field_descriptor{
-            column, value_begin, cudf::io::gpu::seek_field_end(value_begin, end, opts, true)};
+          return field_descriptor{column,
+                                  value_begin,
+                                  cudf::io::gpu::seek_field_end(value_begin, end, opts, true),
+                                  false};
         }();
 
   // Modify start & end to ignore whitespace and quotechars
   auto const trimmed_value_range =
-    trim_whitespaces_quotes(desc_pre_trim.value_begin, desc_pre_trim.value_end, opts.quotechar);
-  return {desc_pre_trim.column, trimmed_value_range.first, trimmed_value_range.second};
+    trim_whitespaces(desc_pre_trim.value_begin, desc_pre_trim.value_end);
+  bool const is_quoted =
```

Comment:
To make @bdice happy :) : `thrust::distance` will return an `int`, not a bool, and you are also doing arithmetic with this after, so maybe just save it as an `int` named `dist_to_first_non_quote` or something explicit.

Reply:
`thrust::distance(..) >=2` is the expression, which returns bool. `thrust::distance` checks the size of the string.
For arithmetic with point, I can static_cast to `ptrdiff_t`.

Reply:
updated.

Reply:
Thanks, changes look right.